lib/ansible/modules/storage/netapp/na_ontap_lun_map.py

```diff
@@ -209,11 +210,13 @@ def get_lun(self):
             lun = result.get_child_by_name('attributes-list').get_child_by_name('lun-info')
 
             # extract and assign lun infomation to return value
+            hexlify = codecs.getencoder('hex')
+            naa_hex = hexlify(str.encode(lun.get_child_content('serial-number')))[0].decode("utf-8")
             return_value = {
                 'lun_node': lun.get_child_content('node'),
                 'lun_ostype': lun.get_child_content('multiprotocol-type'),
                 'lun_serial': lun.get_child_content('serial-number'),
-                'lun_naa_id': '600a0980' + lun.get_child_content('serial-number').encode('hex'),
+                'lun_naa_id': '600a0980' + str(naa_hex),
```

Comment:
see to_text functions in ansible.module_utils._text, they are py2/3 compatible

Reply:
We tried to use to_text, but the feedback I got from our engineer is that codecs was more appropriate here as we are dealing with hexadecimal.

Reply:
AFAIK the Ansible functions should work, could you please provide a sample input value and expected output value given that input.

Reply:
bcoca has said he's happy with this bit